utils/signin.ts

```diff
@@ -0,0 +1,26 @@
+import app from '../firebaseconfig';
+import {
+  UserCredential,
+  signInWithEmailAndPassword,
+  getAuth
+} from 'firebase/auth';
+
+const auth = getAuth(app);
+
+export default async function signIn(
+  email: string,
+  password: string
+): Promise<{ result: UserCredential | null; error: Error | null }> {
+  let result: UserCredential | null = null;
+  let error: Error | null = null;
+  try {
+    result = await signInWithEmailAndPassword(auth, email, password);
+  } catch (e: unknown) {
+    console.log(e);
+    error = e as Error;
+  }
+  if (result != null) {
+    console.log(result.user);
```

Comment:
same with above reasoning, console logging in production build is not recommended unless there is a use for this. If it's needed for something to work, we can keep it.